File: codegen/results/v001/qwen3.5_4b-nvfp4_nothink/20260409_091753/typescript/002_word_frequency/solution.ts
import { readFileSync } from "fs";
import { dirname, join } from "path";

const inputPath = process.argv[2];
if (!inputPath) {
  console.error("Usage: node script.ts <filepath>");
  process.exit(1);
}

// Normalize path to be absolute so it works correctly with Node.js
const absolutePath = join(dirname(inputPath), "input", "text.txt");

const content = readFileSync(absolutePath, "utf-8");

// Convert to lowercase and replace anything that's not a letter with a space
const normalized = content.toLowerCase().replace(/[^a-z]/g, " ");

const words: string[] = normalized.split(" ").filter((w) => w.length > 0);

const freqMap = new Map<string, number>();

for (const word of words) {
  const currentCount = freqMap.get(word) || 0;
  freqMap.set(word, currentCount + 1);
}

// Sort: descending by count, then ascending alphabetically for ties
const sortedWords = Array.from(freqMap.entries())
  .sort((a, b) => {
    if (b[1] !== a[1]) return b[1] - a[1];
    return a[0].localeCompare(b[0]);
  })
  .map(([word, count]) => `${word}: ${count}`);

// Output one line per word
console.log(sortedWords.join("\n"));

File: codegen/results/v001/qwen3.5_4b-nvfp4_nothink/20260409_091753/go/002_word_frequency/solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"regexp"
	"sort"
	"strings"
)

func main() {
	// Open the input file
	file, err := os.Open("input/text.txt")
	if err != nil {
		fmt.Fprintln(os.Stderr, "Error opening file:", err)
		os.Exit(1)
	}
	defer file.Close()

	// Regex to match alphanumeric characters and remove punctuation
	lowercaseWordRe := regexp.MustCompile(`[a-z]+`)
	wordCounter := make(map[string]int)

	// Read and process the file
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		text := scanner.Text()
		// Replace all non-alphabetic characters with spaces, then split into words
		text = lowercaseWordRe.ReplaceAllString(text, " ")
		words := strings.Fields(text)

		for _, word := range words {
			wordCounter[word]++
		}
	}

	if err := scanner.Err(); err != nil {
		fmt.Fprintln(os.Stderr, "Error reading file:", err)
		os.Exit(1)
	}

	// Convert to a slice for sorting
	type wordCount struct {
		word  string
		count int
	}

	var wordsToSort []wordCount
	for word, count := range wordCounter {
		wordsToSort = append(wordsToSort, wordCount{word: word, count: count})
	}

	// Sort by count descending, then alphabetically ascending
	sort.Slice(wordsToSort, func(i, j int) bool {
		if wordsToSort[i].count != wordsToSort[j].count {
			return wordsToSort[i].count > wordsToSort[j].count
		}
		return wordsToSort[i].word < wordsToSort[j].word
	})

	// Output results
	for _, wc := range wordsToSort {
		fmt.Printf("%s: %d\n", wc.word, wc.count)
	}
}